src/main/java/org/corpus_tools/peppermodules/annis/SPointingRelation2ANNISMapper.java

```diff
@@ -115,18 +119,34 @@ public void nodeReached(GRAPH_TRAVERSE_TYPE traversalType,
     // this method behaves exactly as the one in the super class
     super.nodeReached(traversalType, traversalId, currNode, relation, fromNode, order);
     
-    lastEnteredNode = currNode;
+      lastEnteredNode = currNode;
+
+      GlobalIdManager globalIdManager = new GlobalIdManager();
+      boolean escapeCharacters = ((ANNISExporterProperties) this.parentMapper.getProperties()).getEscapeCharacters();
 
-    if (relation != null && relation instanceof SPointingRelation) {
-	  if (relation.getType() != null && !relation.getType().isEmpty()) {
+      if (relation != null && relation instanceof SPointingRelation) {
+        if (relation.getType() != null && !relation.getType().isEmpty()) {
           this.currentComponentName = relation.getType();
-      } else {
-        this.currentComponentName = "salt::NULL";
+        } 
+        else {
+          this.currentComponentName = "salt::NULL";
+        }
+        if (!relation.getLayers().isEmpty()){
+          String layerName = ((SLayer) relation.getLayers().iterator().next()).getName();
+          // check if the name requires escaping
+          String escapedLayerName = globalIdManager.getEscapedIdentifier(layerName);
```

Comment:
Shouldn't the escaping be handled in SRelation2ANNISMapper#mapComponent2ANNIS() ?